### DISCUSSION TITLE
Containers fail on starting v1.7.1

### DISCUSSION THREAD

**Discussion by willyamcts** [nextcloud/all-in-one#1073] [Questions] [answered]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Follow steps in README this repository, run "docker run ..."
2. Visit page starting in port 8080 to the containers status page

### Expected behavior 
Containers installed successfuly and page status all containers "OK"

### Actual behavior 
In page status container Apache and Nextcloud stay status "Starting"

In container nextcloud-aio-nextcloud:
```
Next OC\DB\Exceptions\DbalException: Failed to connect to the database: An exception occurred in the driver: SQLSTATE[08006] [7] connection to server at "nextcloud-aio-database" (172.19.0.5), port 5432 failed: FATAL:  password authentication failed for user "oc_nextcloud" in /var/www/html/lib/private/DB/Exceptions/DbalException.php:71
Stack trace:
#0 /var/www/html/lib/private/DB/ConnectionAdapter.php(59): OC\DB\Exceptions\DbalException::wrap(Object(Doctrine\DBAL\Exception))
#1 /var/www/html/lib/private/legacy/OC_DB.php(55): OC\DB\ConnectionAdapter->prepare('SHOW SERVER_VER...', NULL, NULL)
#2 /var/www/html/lib/private/legacy/OC_DB.php(128): OC_DB::prepare('SHOW SERVER_VER...', NULL, NULL)
#3 /var/www/html/lib/private/legacy/OC_Util.php(759): OC_DB::executeAudited(Array)
#4 /var/www/html/lib/private/legacy/OC_Util.php(739): OC_Util::checkDatabaseVersion()
#5 /var/www/html/lib/private/Console/Application.php(157): OC_Util::checkServer(Object(OC\SystemConfig))
#6 /var/www/html/console.php(99): OC\Console\Application->loadCommands(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#7 /var/www/html/occ(11): require_once('/var/www/html/c...')
#8 {main}installing of nextcloud failed!
Waiting for the database to start...
psql: error: connection to server at "nextcloud-aio-database" (172.19.0.5), port 5432 failed: FATAL:  password authentication failed for user "oc_nextcloud"
Waiting for the database to start...
```

in container nextcloud-aio-database:
```
2022-08-25 23:42:33.359 UTC [38] FATAL:  password authentication failed for user "oc_nextcloud"
2022-08-25 23:42:33.359 UTC [38] DETAIL:  Role "oc_nextcloud" does not exist.
        Connection matched pg_hba.conf line 100: "host all all all scram-sha-256"
```

### Host OS 
Debian 11

#### Nextcloud AIO version 
Last release 1.7.1

#### Current channel 
Latest

#### Others
Same problem reported in issue #1000

**Comment by szaimen** [comment]
Sorry but I cannot reproduce this. Please reset your instance properly by following https://github.com/nextcloud/all-in-one#how-to-properly-reset-the-instance and try again.

**Reply by tanmaywadhwa** [reply]
I am facing the same issue as well. Nextcloud seems to be waiting on the database and when I check database container logs, I see a bunch of 
2022-08-27 23:50:17.582 UTC [173] FATAL:  password authentication failed for user "oc_nextcloud"
2022-08-27 23:50:17.582 UTC [173] DETAIL:  Role "oc_nextcloud" does not exist.

Setting up fresh and I have tried resetting and setting things up again a couple times

**Reply by szaimen** [reply]
So this is a completely new and clean instance (cleaned up with https://github.com/nextcloud/all-in-one#how-to-properly-reset-the-instance)? Or are you each time restoring a backup in the first step after starting a new clean instance?

**Reply by tanmaywadhwa** [reply]
Yup yup. Clean install.

**Reply by szaimen** [reply]
All right! I cannot reproduce this with a clean install as I said before. I just tried it yesterday and it works for me without any problem. 

Unless you find a way for me to reproduce this issue (like exact steps and as much info as possible), I fear there is nothing that I can do to fix it.

**Comment by tanmaywadhwa** [comment]
The only non standard thing I did was to disable all of these before
starting the containers from :8080/containers

ClamAV (only supported on x64, needs ~1GB additional RAM)
Collabora (Nextcloud Office)
Nextcloud Talk (needs ports 3478/TCP and 3478/UDP open in your
firewall/router)
OnlyOffice (only supported on x64)

**Comment by szaimen** [comment]
I guess providing the logs of the database container of the first startup on a clean install would maybe give some clues about why it doesn't work for you.

**Comment by tanmaywadhwa** [comment]
Here are the startup logs. Attaching them here. I am guessing there is a problem with the timezone selection? PST?
[next-cloud-db-logs.txt](https://github.com/nextcloud/all-in-one/files/9456628/next-cloud-db-logs.txt)

**Reply by szaimen** [reply]
Seems like you have tracked down the issue. 

This is clearly a user issue as PST is not even listed as `TZ database name` in the list on wikipedia.

Please try again with a clean instance and e.g. `America/Los_Angeles` and it should work.

**Comment by dcbuser** [comment, marked as answer]
Hi guys, so as noted above it looks like changing timezone on AIO setup screen causes the issue (no idea why).

To confirm, I have created 2 x new Nextcloud servers from scratch. On reaching the domain and optional addon setup screen the timezone is left blank. The Apache and Nextcloud containers are last to start but they reach "running" status within 1 min with no errors in the logs! Hope this helps : )

**Reply by KyuubiYoru** [reply]
Tried to leave timezone blank does't work for me, i get

Waiting for the database to start...
psql: error: connection to server at "nextcloud-aio-database" (172.20.0.3), port 5432 failed: FATAL:  password authentication failed for user "oc_nextcloud"

**Reply by willyamcts** [reply]
> Tried to leave timezone blank does't work for me, i get
> 
> Waiting for the database to start... psql: error: connection to server at "nextcloud-aio-database" (172.20.0.3), port 5432 failed: FATAL: password authentication failed for user "oc_nextcloud"

Try timezone from screen: "Europe/Berlim" if I am not wrong.

**Reply by KyuubiYoru** [reply]
I've tried a few different time zones and none of them work, so I'm just going to stick with my current installation for now. Maybe I'll try again with a later version.

**Reply by szaimen** [reply]
> I've tried a few different time zones and none of them work, so I'm just going to stick with my current installation for now. Maybe I'll try again with a later version.

Can you provide the database container logs if this should still happen after doing a clean install by following https://github.com/nextcloud/all-in-one#how-to-properly-reset-the-instance?

Maybe it helps figuring out why it doesnt work for you

**Reply by tazz4843** [reply]
Hey there, I can still reproduce this issue today, regardless of if I set timezone or not. On Debian 11.5 host, fresh install, i can reproduce this every single time i try a clean install of Nextcloud. Database container logs: https://paste.zero.lgbt/?d513fac6035e4d95#2HsXxEYxVDQhZG3xVz4QaLBifxyA7q2VvgN2CmJzT5EU

cc @szaimen

**Reply by szaimen** [reply]
I don't see an error in the posted logs that could lead to this behaviour... Maybe they are not complete? Also the Nextcloud container logs may be useful...

**Reply by tazz4843** [reply]
That was the entirety of the Postgres logs, just double checked. Anyways here's the Nextcloud container logs: https://paste.zero.lgbt/?7ffde1edbf8681b7#5ZFDjpkAQZazk1YCYAvCjCyEX3KtXBgsyUEH9inKhqRi. Timestamps differ since this is after a complete reinstall, but the Postgres logs are line for line the same, save for timestamps.

**Reply by szaimen** [reply]
Based on the logs there seem to be some files left of an old instance... Did you reset it properly?

**Reply by tazz4843** [reply]
Followed these steps to the letter: https://github.com/nextcloud/all-in-one#how-to-properly-reset-the-instance, and after this (5th?) retry it's still showing the same errors.

**Reply by szaimen** [reply]
Did you configure `NEXTCLOUD_DATADIR` to a different location than the default volume? If yes, you need to clean the target dir as well...

**Reply by tazz4843** [reply]
That appears to have been the issue. Perhaps that should be documented alongside there.

**Reply by szaimen** [reply]
done with https://github.com/nextcloud/all-in-one/commit/0523af0f8a3c8dd98387fe87ffe776b53d79fb23
